OpenRA.Mods.Common/Scripting/Properties/CombatProperties.cs

```diff
@@ -96,7 +96,13 @@ public void Attack(Actor targetActor, bool allowMove = true, bool forceAttack =
 			if (!targetActor.Info.HasTraitInfo<FrozenUnderFogInfo>() && !Self.Owner.CanTargetActor(targetActor))
 				Log.Write("lua", "{1} is not revealed for player {0}!", Self.Owner, targetActor);
 
-			attackBase.AttackTarget(target, true, allowMove, forceAttack);
+			var attack = attackBase.Where(x => !x.IsTraitDisabled && x.HasAnyValidWeapons(target));
+			if (!attack.Any())
+				Log.Write("lua", "Actor {0} was ordered to attack {1} but has no valid Attack* trait to do so!.".F(Self, targetActor));
+			foreach (var a in attack)
+			{
+				a.AttackTarget(target, true, allowMove, forceAttack);
+			}
```

Comment:
Would you prove it, please? I'm only aware of that rule with if blocks, not foreach ones.